fairworkflows/fairstep.py

```diff
@@ -227,6 +239,42 @@ def is_script_task(self, value: bool):
         else:
             self.remove_attribute(RDF.type, object=namespaces.BPMN.ScriptTask)
 
+    @property
+    def code(self):
+        """Returns the code text associated with this FairStep, or None if there isn't any."""
+        stepcode = list(self._rdf.objects(self.code_ref, namespaces.SCHEMAORG.text))
+        if len(stepcode) == 0:
+            return None
+        if len(stepcode) > 1:
+            warnings.warn('There is more than one code text associated with '
+                          'this FairStep in its RDF description: {stepcode}. '
+                          'Returning only the first in list.')
+        return str(stepcode[0])
+
+    @code.setter
+    def code(self, value: str):
+        """Sets the code text associated with this FairStep"""
+        self.set_attribute(namespaces.SCHEMAORG.SoftwareSourceCode, self.code_ref, overwrite=False)
+        self._rdf.add((self.code_ref, namespaces.SCHEMAORG.text, rdflib.Literal(value)))
+
+    @property
+    def programming_language(self):
+        """Returns the programming language for this fairstep's code (either a string, or URI)"""
+        proglang = list(self._rdf.objects(self.code_ref, namespaces.SCHEMAORG.programmingLanguage))
+        if len(proglang) == 0:
+            return None
+        if len(proglang) > 1:
+            warnings.warn('There is more than one programming language associated with the code '
+                          'for this FairStep in its RDF description: {proglang}. Returning only '
+                          'the first in list.')
+        return str(proglang[0])
+
+    @programming_language.setter
+    def programming_language(self, value: str):
+        """Sets the programming language for this fairstep's code (takes a string)"""
+        self.set_attribute(namespaces.SCHEMAORG.SoftwareSourceCode, self.code_ref, overwrite=False)
```

Comment:
I think overwrite should be True? Otherwise we have a lot of duplicate triples after setting programming_language and code properties. (NB this also counts for `code` property)